src/Modules/ModulesGraded.jl

```diff
@@ -2882,3 +2882,51 @@ function _regularity_bound(M::SubquoModule)
   end
   return result
 end
+
+
+###############################################################################
+# Random elements
+###############################################################################
+
+
+function all_partitions_in(n, k)
+    if k == 0
+        return n == 0 ? [Int[]] : []
+    end
+    partitions = Vector{Int}[]
+    for i in 0:n
+        for p in all_partitions_in(n - i, k - 1)
+            push!(partitions, [i; p])
+        end
+    end
+    return partitions
+end
```

Comment:
This should not live here, but in `src/Combinatorics` as it can be applied elsewhere. One day, it can then be replaced by the implementation in `experimental/JuLie/src/partitions.jl`.

Reply:
Yes, I don't mind to shift it and make it available. We will also expand on it. As far as I understood JuLi does partitiions in the sorted sense. This here is what is usually called compositions. So I renamed it.

Reply:
Thanks for moving!
However, I think that this should either include a docstring as well, or shouldn't get exported (or maybe even both for now).

Reply:
For example, changing this later to ZZRingElem instead of Int would be an unnecessary breaking change, so I would rather keep this internal until an interface has been settled. (With talking to people interested in combinatorics like Uli Thiel)